packages/common-sdk/src/web3/transactions/transactions-builder.ts

```diff
@@ -60,17 +48,45 @@ type BaseBuildOption = {
 
 const LEGACY_TX_UNIQUE_KEYS_LIMIT = 35;
 
+/**
+ * A set of options that the builder will use by default, unless overridden by the user in each method.
+ */
+export type TransactionBuilderOptions = {
+  defaultBuildOption: BuildOptions;
+  defaultSendOption: SendOptions;
+  defaultConfirmationCommitment: Commitment;
+};
+
+export const defaultTransactionBuilderOptions: TransactionBuilderOptions = {
+  defaultBuildOption: {
+    maxSupportedTransactionVersion: 0,
+    blockhashCommitment: "confirmed",
+  },
+  defaultSendOption: {
+    skipPreflight: false,
+    preflightCommitment: "confirmed",
+    maxRetries: 3,
+  },
+  defaultConfirmationCommitment: "confirmed",
+};
+
 /**
  * Transaction builder for composing, building and sending transactions.
  * @category Transactions
  */
 export class TransactionBuilder {
   private instructions: Instruction[];
   private signers: Signer[];
+  private opts: TransactionBuilderOptions;
 
-  constructor(readonly connection: Connection, readonly wallet: Wallet) {
+  constructor(
+    readonly connection: Connection,
+    readonly wallet: Wallet,
+    defaultOpts?: TransactionBuilderOptions
+  ) {
     this.instructions = [];
     this.signers = [];
+    this.opts = defaultOpts ?? defaultTransactionBuilderOptions;
```

Comment:
Would it be nicer for users of this class to only be able to pass in partial default options? For example, I don't think the following is possible:

```
const builder = new TransactionBuilder(connection, wallet, {
  defaultBuildOption: {
    blockhashCommitment: "confirmed"
  }
})
```

Reply:
yeah i've been attempting to do that but it's quite difficult to wrestle typescript to have nested partials while having intersection types like we have for BuildOptions.

For now, users would have to build their own by spreading the nested properties in their custom options.